### DISCUSSION TITLE
Help. Redirect is coming back http instead of https in URL

### DISCUSSION THREAD

**Discussion by mk53202** [fief-dev/fief#248] [General]
Making a request for a secured FastAPI endpoint and running into an issue. When I configure the application example in the tutorial after successfully getting it to work fine using a localhost, the redirect for the production deployment passes back the wrong **redirect-uri.**

Resource I'm accessing is: 
`https://myhost.mydomain.com/api/protected
`
All seems to work and the I get a 'redirect_uri is not authorized for this client'. I checked the Redirect URIs in the Admin Console and I have it correctly set as:
`https://myhost.mydomain.com/api/auth-callback
` 
 Looking at the url, it is trying to redirect me to HTTP instead of HTTPS url:

`https://myhost.fief.dev/authorize?response_type=code&client_id=aafpLXpwUCyXE6b2mxs0aDVup66dVG2i3-R23brRx3M&redirect_uri=http%3A%2F%2Fmyhost.mydomain.com%2Fapi%2Fauth-callback&scope=openid`

Any ideas on how to troubleshoot or fix this? Super stumped. Ideas? Please help.

**Comment by frankie567** [comment]
I suppose you're generating the Redirect URI like shown in our [FastAPI web application example](https://docs.fief.dev/integrate/python/fastapi/#web-application-example):

```py
    redirect_uri = request.url_for("auth_callback")
```

If so, it's probably because your FastAPI/Uvicorn server is not aware that it's served behind HTTPS. If you use an hosting platform or proxy that handle HTTPS, you have to set an environment variable so Unicorn accepts it and generates HTTPS URL:

```
FORWARDED_ALLOW_IPS=*
```

Ref: https://www.uvicorn.org/settings/#http

**Reply by AntonOfTheWoods** [reply]
This should be made more explicit. Admittedly, you say "you have to do this", and the problem goes away when you do, but it appears several of us have had the same issue, and an extra warning in the docs (which there are already plenty of! :-) ) could save many of us some time.

**Reply by frankie567** [reply]
There is a dedicated page talking about this ➡️ https://docs.fief.dev/self-hosting/deployment/ssl/

It's the *first result* when searching `https` in the docs search engine:

<img width="1840" alt="Capture d’écran 2023-09-18 à 08 31 10" src="https://github.com/fief-dev/fief/assets/1144727/d5ea093c-d800-4d82-b5c4-85473de24d91">
